File: TypeScript/Codewars/src/resolutions/Are-they-the-same.ts
export function comp(a1: number[] | null, a2: number[] | null): boolean {
	if (!a1 || !a2) return false;
	if (a1.length !== a2.length) return false;

	const aSquared = a1.map((x) => x * x).sort((x, y) => x - y);
	const bSorted = a2.slice().sort((x, y) => x - y);

	return aSquared.every((val, index) => val === bSorted[index]);
}

const a = [121, 144, 19, 161, 19, 144, 19, 11];
const b = [121, 14641, 20736, 361, 25921, 361, 20736, 361];
console.log(comp(a, b));

File: Csharp/CodeWars/src/Completed/AreTheyTheSame.cs
namespace CodeWars.Completed;

public class AreTheyTheSame
{
    public static bool Comp(int[] a, int[] b)
    {
        if (a is null || b is null) return false;
        if (a.Length != b.Length) return false;

        int[] aSquared = [.. a.Select(num => num * num).OrderBy(num => num)];
        int[] bSorted = [.. b.OrderBy(num => num)];

        return !aSquared.Where((val, index) => val != bSorted[index]).Any();
    }
}
